contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6391,6 +6579,690 @@ bbf_ExecDropStmt(DropStmt *stmt)
+	
+	pfree(physical_schema_name);
+	pfree(db_name);
+
+	/* Search for the column specified with partition scheme in table's columns. */
+	tuple = SearchSysCacheAttName(relid, colname);
+
+	/* Raise an error if column specified with partition scheme doesn't exists in table. */
+	if (!HeapTupleIsValid(tuple))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_COLUMN),
+					errmsg("column '%s' does not exist", colname)));
+	}
+
+	attnum = ((Form_pg_attribute) GETSTRUCT(tuple))->attnum;
+	ReleaseSysCache(tuple);
+
+	/* Raise an error if provided partition scheme doesn't exists in the current database. */
+	if (!partition_scheme_exists(dbid, partition_scheme_name))
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+					errmsg("Invalid object name '%s'.", partition_scheme_name)));
+	}
+
+	/* Find the partition scheme used to create partitioned table. */
+	partition_scheme_used_for_table = get_partition_scheme_for_partitioned_table(dbid, logical_schema_name, relname);
+	pfree(logical_schema_name);
+
+	/* partition_scheme_used_for_table will be null for non-partitioned table */
+	if (!partition_scheme_used_for_table ||
+			strcmp(partition_scheme_name, partition_scheme_used_for_table) != 0)
+	{
+		pfree(partition_scheme_name);
+		if (partition_scheme_used_for_table)
+			pfree(partition_scheme_used_for_table);
+		return false;
+	}
+
+	/*
+	 * Column specified with partition scheme should be part of partitioning columns.
+	 * Otherwise, the index is unaligned.
+	 */
+	rel = RelationIdGetRelation(relid);
+	key = RelationGetPartitionKey(rel);
+	partnatts = get_partition_natts(key);
+
+	for (i = 0; i < partnatts; i++)
+	{
+		if (attnum == get_partition_col_attnum(key, i))
+			break;
+	}
+
+	RelationClose(rel);
+
+
+	pfree(partition_scheme_name);
+	if (partition_scheme_used_for_table)
+		pfree(partition_scheme_used_for_table);
+
+	if (i == partnatts) /* not part of partitioning columns */
+		return false;
+	return true;
+}
+
+/*
+ * rename_table_update_bbf_partitions_name
+ *	For a rename operation on a babelfish partitioned table, it renames all partition
+ *	of the table to the new name using hash based on the new name, so that
+ *	name of partitions of any partitioned table doesn't conflict.
+ */
+static void
+rename_table_update_bbf_partitions_name(RenameStmt *stmt)
+{
+	char		*physical_schema_name = stmt->relation->schemaname;
+	char		*logical_schema_name;
+	char		*table_name = stmt->relation->relname;
+	List		*partition_names = NIL;
+	int16		dbid;
+	char		*unique_hash;
+	PlannedStmt	*wrapper;
+	RenameStmt	*rename_partition_stmt = NULL;
+	List		*parsetree;
+	Relation	relation;
+	SysScanDesc	scan;
+	ScanKeyData	key;
+	Oid		parentrelid;
+	Oid		inhrelid;
+	HeapTuple	tuple;
+	const char	*old_dialect;
+	StringInfoData	query;
+	char		*partition_name;
+	char		*new_partition_name;
+
+	logical_schema_name = (char *) get_logical_schema_name(physical_schema_name, true);
+
+	if (!logical_schema_name) /* not a tsql schema */
```

Comment:
Do you mean to use DIALECT check here? If so, then the changes are dialect independent. 
OR
Do you mean to use `tsql_schema_name` instead of `logical_schema_name`? If so, then I think everywhere we follow to use `logical_schema_name`.